File: Seccion 11 - Funciones en C++/Seccion_11_Dart/16_ejercicio_13.dart

import 'dart:io';

List<int> vector = new List();
int tam;

void pedirDatos() {
  int valor;

  stdout.writeln('Digite el tamaño del vector: ');
  tam = int.parse(stdin.readLineSync());

  for (int i = 0; i < tam; i++) {
    stdout.writeln('${i + 1}. Digite un número: ');
    valor = int.parse(stdin.readLineSync());
    vector.insert(i, valor);
  }
}

void cambiarSigno() {
  for (int i = 0; i < tam; i++) {
    vector[i] *= -1;
  }
}

void mostrarVector() {
  print('\nMostrando los elementos del vector con signo cambiado:\n');

  for (int i = 0; i < tam; i++) {
    print(vector[i]);
  }
}

void main() {

  pedirDatos();
  cambiarSigno();
  mostrarVector();

}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/16_ejercicio_13.py

vector = []
tam = 0

def pedir_datos():
    global vector, tam
    
    tam = int(input('Digite el tamaño del vector: '))
    
    for i in range(tam):
        valor = int(input('{}. Digite un número: '.format(i + 1)))
        vector.append(valor)

def cambiar_signo():
    for i in range(tam):
        vector[i] *= -1

def mostrar_vector():
    print('\nMostrando los elementos del vector con signo cambiado:\n')
    
    for i in range(tam):
        print(vector[i])

pedir_datos()
cambiar_signo()
mostrar_vector()